twitter/Models/tweet.m

```diff
@@ -0,0 +1,53 @@
+//
+//  tweet.m
+//  twitter
+//
+//  Created by Nicholas Wayoe on 6/29/20.
+//  Copyright © 2020 Emerson Malca. All rights reserved.
+//
+
+#import "tweet.h"
+#import "User.h"
+@implementation tweet
+-(instancetype) initWithDictionary:(NSDictionary *)dictionary
+{
+    self = [super init];
+    if (self){
+        NSDictionary *originalTweet = dictionary[@"retweeted_status"];
+        if(originalTweet !=nil){
+            NSDictionary *userDictionary = dictionary[@"user"];
+            self.retweetedByUser = [[User alloc] initWithDictionary:userDictionary];
+            dictionary = originalTweet;
+    }
+    self.idStr = dictionary[@"id_str"];
+    self.text = dictionary[@"text"];
+    self.favoriteCount = [dictionary[@"favorite_count"] intValue];
+    self.favorited = [dictionary[@"favorited"] boolValue];
+    self.retweetCount = [dictionary[@"retweet_count"] intValue];
+    self.retweeted = [dictionary[@"retweeted"] boolValue];
+    NSDictionary *user = dictionary[@"user"];
+    self.user = [[User alloc] initWithDictionary:user];
+    
+    NSString *createdAtOriginalString = dictionary[@"created_at"];
+    NSDateFormatter *formatter = [[NSDateFormatter alloc] init];
+    formatter.dateFormat = @"E MMM d HH:mm:ss Z y";
+    NSDate *date = [formatter dateFromString:createdAtOriginalString];
+    formatter.dateStyle = NSDateFormatterShortStyle;
+    formatter.timeStyle = NSDateFormatterNoStyle;
+        
+    self.createdAtString = [formatter stringFromDate:date];
```

Comment:
why not store this as a date and then do formatting in the view. That way different views can use the same date and format differently per their use case.